torchao/quantization/quantize_/workflows/int4/int4_tensor.py

```diff
@@ -113,12 +117,17 @@ def _(func, types, args, kwargs):
         args[1],
         args[2] if len(args) > 2 else None,
     )
+    assert isinstance(weight_tensor, Int4Tensor)
+
     assert weight_tensor.qdata.is_contiguous(), "Expected qdata to be contiguous"
     assert weight_tensor.scale.is_contiguous(), "Expected scale to be contiguous"
     assert weight_tensor.zero_point.is_contiguous(), (
         "Expected zero_point to be contiguous"
     )
 
+    if weight_tensor.activation_scale is not None:
+        input_tensor = input_tensor / weight_tensor.activation_scale
```

Comment:
I think multiplying would be faster, can we change the scale definition to do that?

Reply:
also, how about `input_scale`, to standardize with https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.linear.html

Reply:
actually, maybe `act_scale` or `act_prescale` to standardize with `Float8Tensor.act_quant_kwargs`

if the activation is quantized, do we need to combine the scales to just do the multiplication once? this may also affect the design

Reply:
sure, I can use `act_scale`

> if the activation is quantized, do we need to combine the scales to just do the multiplication once? this may also affect the design

it doesn't work if activation is already quantized, we expect this to be applied before static or dynamic quantization